### PR TITLE
chore(mentions): add integration test for reply approval notification

### PR DESCRIPTION
**Pull request by OrdinaryJellyfish**
**Changes proposed in this pull request:**

**Reviewers should focus on:**

**Screenshot**

**Necessity**

- [ ] Has the problem that is being solved here been clearly explained?
- [ ] If applicable, have various options for solving this problem been considered?
- [ ] For core PRs, does this need to be in core, or could it be in an extension?
- [ ] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).
- [ ] Core developer confirmed locally this works as intended.
- [ ] Tests have been added, or are not appropriate here.

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)
- [ ] Related core extension PRs: (Remove if irrelevant)

### CODE REVIEW

**Review by SychO9 — DISMISSED** [review]

File: extensions/mentions/tests/integration/api/NotificationsTest.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                'json' => [
+                    'data' => [
+                        'attributes' => [
+                            'isApproved' => 1,
+                        ]
+                    ]
+                ]
+            ])
+        );
+    }
```
Comment (askvortsov1) [inline extensions/mentions/tests/integration/api/NotificationsTest.php:101]: Should there be another assert after here?

File: extensions/mentions/tests/integration/api/NotificationsTest.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                ['id' => 4, 'username' => 'acme2', 'email' => 'acme2@machine.local', 'is_email_confirmed' => 1],
+            ],
+            'discussions' => [
+                ['id' => 1, 'title' => __CLASS__, 'created_at' => Carbon::now(), 'last_posted_at' => Carbon::now(), 'user_id' => 1, 'first_post_id' => 1, 'comment_count' => 1, 'last_post_number' => 1, 'last_post_id' => 1],
+                ['id' => 2, 'title' => __CLASS__, 'created_at' => Carbon::now(), 'last_posted_at' => Carbon::now(), 'user_id' => 1, 'first_post_id' => 2, 'comment_count' => 1, 'last_post_number' => 1, 'last_post_id' => 2],
+
+                ['id' => 33, 'title' => __CLASS__, 'created_at' => Carbon::now(), 'last_posted_at' => Carbon::now(), 'user_id' => 1, 'first_post_id' => 33, 'comment_count' => 6, 'last_post_number' => 6, 'last_post_id' => 38],
+            ],
+            'posts' => [
+                ['id' => 1, 'discussion_id' => 1, 'created_at' => Carbon::createFromDate(1975, 5, 21)->toDateTimeString(), 'user_id' => 1, 'type' => 'comment', 'content' => '<t><p>foo bar</p></t>', 'number' => 1],
```
Comment (askvortsov1) [inline extensions/mentions/tests/integration/api/NotificationsTest.php:40]: Do we need this much loading data?

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

File: extensions/mentions/tests/integration/api/NotificationsTest.php (hunk 1 above)
Comment (SychO9) [reply extensions/mentions/tests/integration/api/NotificationsTest.php:101]: an `$this->assertEquals(1, $mainUser->getUnreadNotificationCount());` is what will confirm this works.

File: extensions/mentions/tests/integration/api/NotificationsTest.php (hunk 2 above)
Comment (SychO9) [reply extensions/mentions/tests/integration/api/NotificationsTest.php:40]: we can delete the `33` discussion and posts from `33`+

File: extensions/mentions/tests/integration/api/NotificationsTest.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+        $this->assertEquals(0, $mainUser->getUnreadNotificationCount());
+
+        $response = $this->send(
+            $this->request('POST', '/api/posts', [
+                'authenticatedAs' => 4,
+                'json' => [
+                    'data' => [
+                        'attributes' => [
+                            'content' => 'reply with predetermined content for automated testing - too-obscure',
```
Comment (SychO9) [inline extensions/mentions/tests/integration/api/NotificationsTest.php]: To test the reply notification works let's add a post mention (post mention = replying to a post)
we should try to mention a post created by `mainUser` (userId = 2) so let's do `@"mainUser"#p2`
and change the `user_id` of post `id 2` to `2`

File: extensions/mentions/tests/integration/api/NotificationsTest.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                ]
+            ])
+        );
+
+        $this->assertEquals(0, $mainUser->getUnreadNotificationCount());
+
+        $json = json_decode($response->getBody()->getContents(), true);
+
+        $this->send(
+            $this->request('PATCH', '/api/posts'.$json['data']['id'], [
```
Comment (SychO9) [inline extensions/mentions/tests/integration/api/NotificationsTest.php]:
```suggestion
            $this->request('PATCH', '/api/posts/'.$json['data']['id'], [
```

**Review by SychO9 — APPROVED** [review]

**Pull request merged**